svgdigitizer/electrochemistry/electrolyte.py

```diff
@@ -0,0 +1,196 @@
+            {'CID': 1004, 'InChI': 'InChI=1S/H3O4P/c1-5(2,3)4/h(H3,1,2,3,4)', 'InChIKey': 'NBIIXXVUZAFLBC-UHFFFAOYSA-N'}
+
+        """
+        results = pcp.get_cids(chemical_name, 'name')
+        len_results = len(results)
+        if len_results > 1:
+            raise RuntimeError("Ambiguous chemical name! More than one chemical id return by pubchem. Use another chemical name")
+        elif len_results == 0:
+            raise RuntimeError(f"Name of chemical {chemical_name} not found in pubchem! Check spelling.")
+        else:
+            return pcp.get_properties(("InChIKey", "InChI"), results[0])[0]
+
+    def get_pKa(self, InChIKey):
+        r"""
+        Get pKa
+        Pubchem lookup only works for neutral acids!
+
+        """
+        pattern = re.compile(r'.*(?P<pKa>\d+\.\d*).*')
+        try:
+            return acid_base_dict[InChIKey]['components']
+        except KeyError:
+            cid = self.lookup_CID_from_InChI(InChIKey=InChIKey)
+            pKa_result = self.lookup_pKa(cid)['pKa']
+            # TODO support salts etc.
+            return [Acid(pKa=[pattern.match(i).group('pKa') for i in pKa_result.split(';')], charge=0)]
+
+    def lookup_pKa(self, cid):
+        r"""
+        get url from Pubchem to get pka lookup result
+        'XML' can be replaced with 'JSON' but it is harder to parse later on
+        for more info about Pubchem output types: https://pubchemdocs.ncbi.nlm.nih.gov/pug-rest$_Toc494865558
+        stolen from https://github.com/khoivan88/pka_lookup/blob/master/src/pka_lookup_pubchem.py
+        """
+        pka_lookup_result_xml = f'https://pubchem.ncbi.nlm.nih.gov/rest/pug_view/data/compound/{cid}/XML?heading=Dissociation+Constants'
+        headers = {
+            'user-agent': 'Mozilla/5.0 (X11; CentOS; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.75 Safari/537.36'}
+
+        # Get the html request info using CID number from pubchem
+        r = requests.get(pka_lookup_result_xml, headers=headers, timeout=15)
+        # Check to see if give OK status (200) and not redirect
+        if r.status_code == 200 and len(r.history) == 0:
+            # print(r.text)
+            # Use python XML to parse the return result
+            tree = ET.fromstring(r.text)
+
+            # Get the XML tree of <Information> only
+            info_node = tree.find('.//*{http://pubchem.ncbi.nlm.nih.gov/pug_view}Information')
+
+            # Get the pKa reference:
+            original_source = info_node.find('{http://pubchem.ncbi.nlm.nih.gov/pug_view}Reference').text
+            # Get the pKa result:
+            pka_result = info_node.find('.//*{http://pubchem.ncbi.nlm.nih.gov/pug_view}String').text
+            pka_result = re.sub(r'^pKa = ', '', pka_result)    # remove 'pka = ' part out of the string answer
+
+            result = {
+                'pKa': pka_result,
+                'reference': original_source,
+            }
+
+            return result
```

Comment:
```suggestion
            return pka_result
```
I don't think we are using any of the other data.